### ISSUE TITLE
Feature request: custom rates for custom time ranges

### ISSUE THREAD

**Issue by flogeld** [kevinpapst/kimai2#309]
Right now kimai2 feels like it is made for freelancers only. With the following additions I would love to use it to track working hours of any company's employees.

I would like to track working hours project unrelated and with custom rates for e.g. nightshifts.

In a monthly report I would then like to see how much time each employee worked with the different rates displayed seperately.

In my opinion this would make kimai2 a really universial time tracking software for any business.

**Comment by kevinpapst** [comment]
Hi!

You can configure different rate factors for night shifts: 
https://github.com/kevinpapst/kimai2/blob/master/var/docs/configurations.md#hourly-rates-for-timesheet-records

A project does always relate to a customer. Maybe the customer is just another department in your own company, but it exists. 
What will be changed: there will be global activities, you can follow the PR #259 if you are interested.

With the invoice updates in #306 you will be export monthly user reports. 
I'll see if I can add a simple Calculator that groups by user and hourly rate in the same PR.

Thanks for your input!

**Comment by flogeld** [comment]
Thank you for your response. But how can I set a custom rate for e.g. 22pm to 6am? Can I pass the array other arguments than the weekdays?

**Comment by kevinpapst** [comment]
True, I mixed that up. You can't right now, you can only set different rates for different weekdays.

**Comment by flogeld** [comment]
But how can a night shift be different from a day shift then?
Will this get implemented?

**Comment by kevinpapst** [comment]
Yes, it will be implemented in the future.
Get others to vote for that feature, so its importance will be increased.
For now I would recommend to create another activity with a higher hourly rate.

**Comment by kabums** [comment]
I vote for this feature ! We also nedd different rates for 06:00pm to 06:00am.

**Comment by JujanS** [comment]
> I vote for this feature ! We also nedd different rates for 06:00pm to 06:00am.

You have a solution ?

**Comment by rbuehler-teletrend** [comment]
I vote for this feature, too.
It would be great if you can set a different rate factor for early morning hours as well as late evening hours.
